### DISCUSSION TITLE
Generate custom link for change email confirmation

### DISCUSSION THREAD

**Discussion by haydn** [supabase/supabase#5125] [Questions] [answered]
We're generating all our own emails and injecting links created with the `generateLink()` function. (This is so we can fully personalise the emails and so they match the other emails we send out.) However, I'm not sure how to get this working for the confirmation emails sent when a user's email address is changed. It seems like those emails are only ever sent as a side-effect of changing a user's email with `updateUser()` and the only way to customise those emails is via config.

So, my questions are:

1. How do we generate the custom links for email change confirmations?
2. How do we prevent the default emails from being sent out when a user is updated?

**Comment by kangmingtay** [comment, marked as answer]
Hi @haydn, 

1. Currently, we don't support generating custom links for email change confirmations. But I'll create a github issue on gotrue to track it and look to add it soon. (https://github.com/supabase/gotrue/issues/361)
2. Could you elaborate more on when and how this happens?

**Reply by haydn** [reply]
Hey @kangmingtay,

1. Okay, good to confirm that it is not currently supported and haven't just misunderstood.
2. My original point 2 probably isn't relevant if there's no way to generate custom links for email address changes. I'll add some thoughts onto that GoTrue issue about what would be a good API for our use case.

Thanks!

**Comment by EnzoPujol** [comment]
Any updates in this issue? I need the same functionality
